### ISSUE TITLE
Remove name consistency enforcement for `NewType` and `TypedDict`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7113]
After some discussion in the [typing discussion forum](https://discuss.python.org/t/draft-of-typing-spec-chapter-for-enums/43496/15
), it was generally agreed that enforcing name consistency for `NewType` and `TypedDict` (functional form) should not be the job of a type checker.

The following two lines should therefore not generate an error.

```python
OtherTD = TypedDict("TD", {"a": int})
OtherNT = NewType("NT", int)
```

**Comment by erictraut** [comment]
The spec allows reporting an error in this case, and mypy does issue errors for these, so I'm going to leave this check in place.
